tests/unit/test_mac.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include "util.cpp"
+
+#include "mococrw/mac.h"
+#include "mococrw/hash.h"
+
+using namespace mococrw;
+
+using testing::Eq;
+
+struct inputData
+{
+    std::string key;
+    std::string data;
+    size_t outputLength;
+    std::string expectedRestultSha256;
+    std::string expectedRestultSha384;
+    std::string expectedRestultSha512;
+
+};
+
+void testHmacSha(openssl::DigestTypes hashFunction, inputData testData, std::string expectedResult)
+{
+    auto hmac = mococrw::HMAC(hashFunction, utility::fromHex(testData.key), testData.outputLength);
+    hmac.update(utility::fromHex(testData.data));
+    ASSERT_THAT(utility::toHex(hmac.finish()),
+                Eq(expectedResult));
+}
+
+void testHmac(struct inputData testData)
+{
+    testHmacSha(openssl::DigestTypes::SHA256, testData, testData.expectedRestultSha256);
+    testHmacSha(openssl::DigestTypes::SHA384, testData, testData.expectedRestultSha384);
+    testHmacSha(openssl::DigestTypes::SHA512, testData, testData.expectedRestultSha512);
+}
+
+TEST(hmac, tests)
```

Comment:
I think this might also be a good place to use parametrized tests using `TEST_P` if you move the parameters and expected result into a struct.